### PR TITLE
Update doc on mounts

### LINKED ISSUE #3531: Mounts: "shared" type
### Where on docs.platform.sh should be changed?

TBD

### What exactly should be updated?

TBD

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AnouckColson**
## Why

Closes #3531 

## What's changed

- Added `storage` and `tmp` types to app reference page (only `tmp` for PSH docs) & updated content on mounts to reflect current status.
- Updated Network Storage page

### CODE REVIEW

**Review by lolautruche — DISMISSED** [review]

File: sites/platform/src/create-apps/app-reference.md [hunk 1]
```diff
@@ -297,89 +302,110 @@ The `{{< variable "DIRECTORY" >}}` is relative to the [app's root](#root-directo
...
 | `service`     | `string`             |          | The name of the [network storage service](../add-services/network-storage.md). |
 
-Basic example:
+<--->
+<!-- Upsun -->
+
+| Name          | Type                 | Required | Description |
+| ------------- | -------------------- | -------- | ----------- |
+| `source`      | `storage`, `service`, or `tmp` | Yes      | Specifies the mount type. Set to: </br> - `storage` so your mount can be shared between instances of the same app.</br> - `service` so your mount can be shared between several apps.</br> - `tmp` to mount a directory within the `/tmp` directory of your app.`tmp` mounts are local ephemeral mounts that are removed when the app container is moved to another host. They allow you to store files that you're not afraid to lose, such as your application cache, without having to create a symlink to `/tmp` in your build hook.</br>Note that the  `/tmp` directory has a [maximum allocation of 4 GB](/create-apps/troubleshoot-disks.md#no-space-left-on-device). If your project requires more disk, [contact Support](/learn/overview/get-support.md).  |
```
Comment (lolautruche) [inline sites/platform/src/create-apps/app-reference.md:318]:
```suggestion
| `source`      | `storage`, `service`, or `tmp` | Yes      | Specifies the mount type. Set to: </br> - `storage` so your mount can be shared between instances of the same app.</br> - `service` so your mount can be shared between several apps (using the Network Storage service).</br> - `tmp` to mount a directory within the `/tmp` directory of your app.`tmp` mounts are local ephemeral mounts; their content may be removed during infrastructure maintenance operations. They allow you to store files that you're not afraid to lose, such as your application cache, without having to create a symlink to `/tmp` in your build hook.</br>Note that the  `/tmp` directory has a [maximum allocation of 8 GB](/create-apps/troubleshoot-disks.md#no-space-left-on-device). If your project requires more disk, [contact Support](/learn/overview/get-support.md).  |
```

☝️ Probably needs link to the network storage service doc page.

File: sites/platform/src/create-apps/app-reference.md [hunk 2]
```diff
@@ -297,89 +302,110 @@ The `{{< variable "DIRECTORY" >}}` is relative to the [app's root](#root-directo
...
 
 {{% version/only "2" %}}
 
 ### Mounts, instances, and Network Storage
 
-Even though the configuration is slightly different for `local` and `service` (Network Storage) mounts,
-in reality **all** {{% vendor/name %}} mounts are [Network Storage](/add-services/network-storage.md) mounts.
+On {{% vendor/name %}}, `storage` mounts are widely used as they can be shared between different instances of the same app,
+which enables [horizontal scaling](/manage-resources/_index.md).
+Under the hood, `storage` mounts are actually [Network Storage](/add-services/network-storage.md) services.
```
[the remark this answers is not in the document]
Comment (AnouckColson) [reply sites/platform/src/create-apps/app-reference.md:391]: I hear you, I've tried to simplify it.

File: sites/platform/src/create-apps/app-reference.md [hunk 3]
```diff
@@ -297,89 +302,110 @@ The `{{< variable "DIRECTORY" >}}` is relative to the [app's root](#root-directo
...
-```
+If you need a local mount, {{% vendor/name %}} allows you to mount a directory within the `/tmp` directory of your app.
+However, keep the following limitations in mind:
 
-Remember that the `/tmp` has a [maximum allocation of 4 GB](/create-apps/troubleshoot-disks.md#no-space-left-on-device). 
-If your project requires more disk, [contact Support](/learn/overview/get-support.md).
+- `tmp` mounts are removed when your app container is moved to another host.
+- The `/tmp` directory has a [maximum allocation of 4 GB](/create-apps/troubleshoot-disks.md#no-space-left-on-device).
+  You can [ask Support](/learn/overview/get-support.md) for more disk,
+  but 8 GB is the maximum that can be allocated to `/tmp`.
```
[the remark this answers is not in the document]
Comment (AnouckColson) [reply sites/platform/src/create-apps/app-reference.md]: Double checked and you're right, max is already 8 GB. I removed the last sentence as I don't think we wan't to encourage users to ask for more.

File: sites/platform/src/create-apps/app-reference.md [hunk 4]
```diff
@@ -297,89 +302,110 @@ The `{{< variable "DIRECTORY" >}}` is relative to the [app's root](#root-directo
...
 
 {{% version/specific %}}
+<!-- Platform.sh -->
 ```yaml {configFile="app"}
-mounts:
-    'web/uploads':
-        source: local
-        source_path: uploads
+    mounts:
+      '/.tmp_platformsh':
```
[the remark this answers is not in the document]
Comment (AnouckColson) [reply sites/platform/src/create-apps/app-reference.md:333]: Committed

File: sites/platform/src/create-apps/app-reference.md [hunk 5]
```diff
@@ -297,89 +302,110 @@ The `{{< variable "DIRECTORY" >}}` is relative to the [app's root](#root-directo
...
+
 The accessibility to the web of a mounted directory depends on the [`web.locations` configuration](#web).
 Files can be all public, all private, or with different rules for different paths and file types.
 
 Note that when you back up an environment,
 the mounts on that environment are backed up too.
 
 Also, mounted directories aren't deleted when they're removed from `{{< vendor/configfile "app" >}}`.
-The files still exist on disk until manually removed.
+The files still exist on disk until manually removed (or, for `tmp` mounts, until the app container is moved to another host).
```
[the remark this answers is not in the document]
Comment (AnouckColson) [reply sites/platform/src/create-apps/app-reference.md]: Applied

File: sites/platform/src/create-apps/app-reference.md [hunk 6]
```diff
@@ -297,89 +302,110 @@ The `{{< variable "DIRECTORY" >}}` is relative to the [app's root](#root-directo
...
 ### Mounts, instances, and Network Storage
 
-Even though the configuration is slightly different for `local` and `service` (Network Storage) mounts,
-in reality **all** {{% vendor/name %}} mounts are [Network Storage](/add-services/network-storage.md) mounts.
+On {{% vendor/name %}}, `storage` mounts are widely used as they can be shared between different instances of the same app,
+which enables [horizontal scaling](/manage-resources/_index.md).
+Under the hood, `storage` mounts are actually [Network Storage](/add-services/network-storage.md) services.
 
-With this in mind, there are some consequences to be aware of:
+If you want data in a mount to be shared between applications,
```
[the remark this answers is not in the document]
Comment (AnouckColson) [reply sites/platform/src/create-apps/app-reference.md]: Applied

File: sites/platform/src/create-apps/app-reference.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (AnouckColson) [reply sites/platform/src/create-apps/app-reference.md]: Applied

File: sites/platform/src/create-apps/app-reference.md [hunk 7]
```diff
@@ -297,89 +302,110 @@ The `{{< variable "DIRECTORY" >}}` is relative to the [app's root](#root-directo
...
 
-Remember that the `/tmp` has a [maximum allocation of 4 GB](/create-apps/troubleshoot-disks.md#no-space-left-on-device). 
-If your project requires more disk, [contact Support](/learn/overview/get-support.md).
+- `tmp` mounts are removed when your app container is moved to another host.
+- The `/tmp` directory has a [maximum allocation of 4 GB](/create-apps/troubleshoot-disks.md#no-space-left-on-device).
+  You can [ask Support](/learn/overview/get-support.md) for more disk,
+  but 8 GB is the maximum that can be allocated to `/tmp`.
 
-{{% /note %}}
+Therefore, `tmp` mounts are ideal to store non-critical data, such as your application cache,
```
[the remark this answers is not in the document]
Comment (AnouckColson) [reply sites/platform/src/create-apps/app-reference.md]: Applied

**Review by chadwcarlson — APPROVED** [review]
1) The resource-init page won't exist until https://github.com/platformsh/platformsh-docs/pull/3579 is merged, so fixing that link to allow for successful build.
2) The indents were inconsistent, so quick fixes there.

Otherwise good to go on my end!

**Pull request merged**